### ISSUE TITLE
Dates transfromed into Maps

### ISSUE THREAD

**Issue by ryanking1809** [alloc/wana#13]
Looks like observable dates a transformed into maps, breaking some of the functionality.

```js
let date = o(new Date()) // makes a Map proxy
console.log(new Date() - new Date()); // 0
console.log(count.date - count.date); // TypeError: this is not a Date object
```

Here's something I hacked together using the deepObervable class
https://codesandbox.io/s/wana-deepobservable-class-sckdz

**Comment by aleclarson** [comment]
I haven't been using `Date` in my apps. I typically use `Date.now` instead. ;)

Fixed in v0.10.4

The following built-in types are never wrapped by the `o` function:
- Date
- RegExp
- Promise
- Generator
- GeneratorFunction
- AsyncFunction
- WeakMap
- WeakSet

**Comment by ryanking1809** [comment]
Yeah I suppose I can just store it as a number and convert it when I need
it to be a Date :)

Thanks!
